ideascube/tests/helpers.py

```diff
@@ -0,0 +1,39 @@
+from django.apps import apps
+from django.test import TransactionTestCase
+from django.db.migrations.executor import MigrationExecutor
+from django.db import connection
+import pytest
+
+needMigration = pytest.mark.skipif(
+    pytest.config.getoption("--nomigrations"),
+    reason="need migrations to run tests on migrations"
+)
+
+@needMigration
+class MigrationTestCase(TransactionTestCase):
+    '''A Test case for testing migrations'''
+
+    # These must be defined by subclasses.
+    migrate_from = None
+    migrate_to = None
+
+    def setUp(self):
+        super(MigrationTestCase, self).setUp()
+
+        self.executor = MigrationExecutor(connection)
+        self.executor.loader.project_state(self.migrate_from).apps
+        self.executor.loader.build_graph()  # reload.
+        self.executor.migrate(self.migrate_from)
+
+        self.setUpDataBeforeMigration()
+
+        self.executor.loader.build_graph()  # reload.
+        self.executor.migrate(self.migrate_to)
```

Comment:
So if I get this right, the way this works is that if I want to test a migration I would do this:

```python
class SomeTest(MigrationTestCase):

    migrate_from = ...
    migrate_to = ...

    def setUpDataBeforeMigration(self):
        # Add some data
        ...

    def test_something(self):
        # Just check that the migration went fine
        ...
```

But from what I understand, if I want to run multiple tests on the same migration, with different pre-loaded data, I'd have to do:

```python
class SomeTest(MigrationTestCase):

    migrate_from = ...
    migrate_to = ...

    def setUpDataBeforeMigration(self):
        # Add some data
        ...

    def test_something(self):
        # Just check that the migration went fine
        ...


class SomeOtherTest(MigrationTestCase):

    migrate_from = ...  # Same as previous class
    migrate_to = ...  # Same as previous class

    def setUpDataBeforeMigration(self):
        # Add some different data
        ...

    def test_something(self):
        # Just check that the migration went fine
        ...


...
```

That is, once test case class per test.

I'd prefer being able to write only one test case class for **all the tests on a given migration**:

```python
class MigrationXXXTestCase(MigrationtestCase):

    migrate_from = ...
    migrate_to = ...

    def test_something(self):
        # Set up the data here
        ...

        # Run the migration here
        ...

    def test_something_else(self):
        ...

    def test_a_third_thing(self):
        ...
```

That does imply a slight change on the API exposed by your `MigrationTestCase` class:

* the `setupDataBeforeMigration` method becomes unnecessary, each test function is responsible for loading its data;
* you need a new `run_migration` method which each test function can call (it would probably just run the last two `build_graph` and `migrate` lines)

What do you think?